common/lib/dependabot/dependency_grapher/generic.rb

```diff
@@ -0,0 +1,69 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+require "dependabot/dependency_grapher/base"
+
+module Dependabot
+  module DependencyGrapher
+    class Generic < Base
```

Comment:
As noted in the description, this isn't typical for Dependabot and I think in order to properly assign `purl_pkg_for` keys to the right area of the code - otherwise this class would need to hold implicit understanding of which ecosystems don't have their own class.